xdsl/traits.py

```diff
@@ -197,6 +197,73 @@ def verify(self, op: Operation) -> None:
                         regions += child_op.regions
 
 
+class SymbolTable(OpTrait):
+    """
+    SymbolTable operations are containers for Symbol operations. They offer lookup
+    functionality for Symbols, and enforce unique symbols amongst its children.
+
+    A SymbolTable operation is constrained to have a sing-block single region.
+    """
+
+    def verify(self, op: Operation):
+        # import builtin here to avoid circular import
+        from xdsl.dialects.builtin import StringAttr
+
+        if len(op.regions) != 1:
+            raise VerifyException(
+                "Operations with a 'SymbolTable' must have exactly one region"
+            )
+        if len(op.regions[0].blocks) != 1:
+            raise VerifyException(
+                "Operations with a 'SymbolTable' must have exactly one block"
+            )
+        block = op.regions[0].block
```

Comment:
```suggestion
        block = op.regions[0].blocks[0]
```
the block helper does the checking for single block again, seems unnecessary